pandas/core/series.py

```diff
@@ -1282,11 +1282,6 @@ def _set_with(self, key, value):
 
             if is_scalar(key):
                 key = [key]
-            elif not isinstance(key, (list, Series, np.ndarray)):
-                try:
-                    key = list(key)
```

Comment:
note below on 1291 we check for Index, which the check on 1285 doesnt do.  The only time we actually get to 1287 in the tests is with a DatetimeIndex, which we should _not_ be casting to list.